core-backend/wasmtime/src/env.rs

```diff
@@ -334,7 +324,24 @@ where
 
         let termination_reason = self.store.data().termination_reason.clone();
 
-        let (info, wasm_memory_addr) = self.prepare_post_execution_data()?;
+        let wasm_memory_addr = self.get_wasm_memory_begin_addr();
+        let WasmtimeEnvironment {
+            mut store,
+            ext,
+            memory,
+            ..
+        } = self;
+        let (info, _) = ext
+            .into_inner()
+            .into_ext_info(|offset: usize, buffer: &mut [u8]| {
+                memory.read(&mut store, offset, buffer)
+            })
+            .map_err(|(reason, gas_amount)| BackendError {
+                reason: WasmtimeEnvironmentError::MemoryAccess(reason),
+                description: None,
+                gas_amount,
+            })
+            .map(|info| (info, wasm_memory_addr))?;
```

Comment:
I don't know what the way you're gonna go however the map in this case is unnecessary.